docs/Technical-Documentation/aws-es-proxy.md

```diff
@@ -0,0 +1,12 @@
+# AWS Elastic Search Proxy
+
+## Background
+TDP leverages the Elastic stack to provide fast search capabilities along with powerful visualizations to provide OFA insight into the data collected from STTs. TDP deploys it's own Kibana instance and relies on Cloud.gov and AWS to provides Elastic Search as a service. Because TDP leverages Elastic Search as a service, the app is subject to constraints imposed by this choice.
+
+## AWS ES Proxy
+TDP leverages Elastic Search as a service through Cloud.gov/AWS. However, AWS has specific requirements to accept requests flowing to their services. That is, all requests coming to an AWS hosted service must be signed according to the [AWS Signature Version 4](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html) signing method. Therefore, when deploying your own services that need to connect to an AWS service (Kibana in our case), you must also deploy/introduce a proxy application that will sign your app's requests appropriately. AWS provides [this resource](https://aws.amazon.com/blogs/security/how-to-control-access-to-your-amazon-elasticsearch-service-domain/) on how an app can have it's requests signed appropriately. Using this resource, the engineering team introduced the [AWS ES Proxy](https://github.com/raft-tech/aws-es-proxy) which is a V4 signing proxy forked off of [this proxy](https://github.com/abutaha/aws-es-proxy). The engineering team forked this repo to allow for the modification of ports that the proxy opens during it's deployment.
```

Comment:
@elipe17 can you speak to the security implications of having this image managed in DockerHub, since I assume this image is used in all environments? (ACF Tech has raised concerns about this in the past). 

And I assume, this is one of the resources that any engineering team working on the TDP project in the future would have to set up (or the OFA team would have to)?

Reply:
The docker hub we used is owned by raft. And as far as security implications, there is no credentials saved in the docker file so in the event if someone has access to the hub, there will be no security issue.
However, if TANF owns a docker hub, we can definitely use that hub.

Reply:
@raftmsohani can you add your comment to the markdown? then feel free to merge. thank you!